public/app/features/dashboard/components/DashNav/IirisMaintenanceListModal.tsx

```diff
@@ -76,7 +85,7 @@ export class IirisMaintenanceListModal extends React.PureComponent<Props, State>
         );
       })
       .then((maintenances: Maintenance[]) => {
-        this.setState({ allMaintenances: maintenances });
+        this.setState({ isLoading: false, allMaintenances: maintenances });
```

Comment:
What if there has been error in fetching the maintenances? Now the loading icon stays there. I guess the user can then close the dialog and try to open it again?

Reply:
Well that is one possibility. I've seen perpetual loops that have couple of seconds in between loading in these cases, so I guess that is one approach. Should I make one? Basically we make three states, loading, finished and error. We show a spinner when loading, maintenances when finished and a message "re-trying after n-seconds..." when an error has occured.